packages/squiggle-lang/src/dist/distOperations/binaryOperations.ts

```diff
@@ -51,3 +56,48 @@ export const binaryOperations = {
   pointwiseLogarithm: (t1, t2, { env }) => pointwise(t1, t2, env, "Logarithm"),
   pointwisePower: (t1, t2, { env }) => pointwise(t1, t2, env, "Power"),
 } satisfies { [k: string]: BinaryOperation };
+
+export const algebraicSum = (dists: BaseDist[], env: Env): DistResult =>
+  dists.reduce<DistResult>(
+    (accumulatedDist, currentDist) =>
+      bind(accumulatedDist, (aVal) =>
+        binaryOperations.algebraicAdd(aVal, currentDist, { env })
+      ),
+    Ok(new PointMass(0))
+  );
+
+export const algebraicProduct = (dists: BaseDist[], env: Env): DistResult =>
+  dists.reduce<DistResult>(
+    (accumulatedDist, currentDist) =>
+      bind(accumulatedDist, (aVal) =>
+        binaryOperations.algebraicMultiply(aVal, currentDist, { env })
+      ),
+    Ok(new PointMass(1))
+  );
+
+export const algebraicCumSum = (dists: BaseDist[], env: Env): BaseDist[] =>
+  accumulate(dists, (a, b) =>
+    getExt(
```

Comment:
It would have been nicer to return the error, but this would have been a fair bit more work.